### PR TITLE
Update replication manager to use new code

### PR DESCRIPTION
**Pull request by nickva**
First commit is a clean-up / refactor to prepare to switch to new scheduler. After this refactor replications still work, eunit tests still pass.

Changes:
- Code from replication manager which deals with doc updating moved to
  `couch_replicator_docs.erl`
- Code dealing with calculating replication ids moved to `couch_replicator_ids.erl`
- Few functions left in `_utils.erl`, but kept external exported functions
  as proxied to avoid modifying various unlreated modules which call them.
- Added TODO: comments to functions which will need attention later:
  - Still using old code
  - Unused functions
  - `utils` function but used from only one place, etc.

EUnit test still pass:
`make eunit skip_deps+=couch_epi,couch_log apps=couch_replicator`

```
=======================================================
  All 166 tests passed.
==> rel (eunit)
==> couchdb (eunit)
```

Replication utility from dyno still show docs trigger and replicate:

```
rep.replicate_1_to_n_then_check_replication(1)
 > created  1 dbs with prefix rdyno_src_
 > created  1 dbs with prefix rdyno_tgt_
updating documents
 > _replicator rdyno_0001_0001 : 7-1aac3a47cacc5dd2f5cf27cc3d603e49
waiting for replication documents to trigger
 > retrying function wait_to_trigger
 > function wait_to_trigger succeded after 10.029 +/- 10.0  sec.
all replication documents triggered
>>> update cycle 0  <<<
 > waiting for target  1
 > function wait_till_dbs_equal succeded after 0.024 +/- 1.0  sec.
 > waiting to propagate changes from  1 to 1  : 0.042 sec.
```

Next commit is a enables adding jobs to scheduler, changes replicator manager to be 3 separate gen_servers (clustering, doc processing/management, replicator doc change feeds listener).

In this commit, old eunit tests still pass but rep test doesn't work anymore.

### CODE REVIEW

File: src/couch_multidb_changes.erl [hunk 1]
```diff
@@ -268,19 +268,11 @@ scan_all_dbs(Server, DbSuffix) when is_pid(Server) ->
...
-        _ ->
-            false
-    end.
-
 is_design_doc({Change}) ->
     case lists:keyfind(<<"id">>, 1, Change) of
         false ->
             false;
-        {Id, _} ->
+        {_, Id} ->
```
Comment (iilyak) [inline src/couch_multidb_changes.erl:275]: Good catch

File: src/couch_multidb_changes.erl [hunk 2]
```diff
@@ -93,7 +93,7 @@ handle_call({change, DbName, Change}, _From,
     case {SkipDDocs, is_design_doc(Change)} of
         {true, true} ->
             {reply, ok, State};
-        {false, _} ->
```
Comment (rnewson) [inline src/couch_multidb_changes.erl:96]: explain?

Comment (nickva) [inline src/couch_multidb_changes.erl:96]: Covers the case where skip_ddocs is `true` but is_design_doc(Change) is `false`. Basically we want to call the change callback in every case except `{true, true}`

**Comment by rnewson** [comment]
+1

**Pull request merged**